src/system/decorators/resolver.ts

```diff
@@ -2,6 +2,7 @@ import { Uri } from 'vscode';
 
 function replacer(key: string, value: any): any {
 	if (key === '') return value;
+	if (key == 'container') return undefined;	// container may contain circular references.
```

Comment:
For this one -- I tried making this change, but couldn't get it to hit this case -- when are you seeing it? I ask because this isn't really the right fix -- it implies that something is passing a rich object through a decorator, which shouldn't be happening.

Reply:
Removed this for now. Will open a new issue if I can reproduce this.